src/dbnode/storage/namespace.go

```diff
@@ -602,6 +602,13 @@ func (n *dbNamespace) QueryIDs(
 		n.metrics.queryIDs.ReportError(n.nowFn().Sub(callStart))
 		return index.QueryResults{}, errNamespaceIndexingDisabled
 	}
+
+	if n.reverseIndex.BootstrapsDone() < 1 {
```

Comment:
interesting...wasn't this handled at the RPC level before? Whats the change

Reply:
I think I found that actually the query service when testing shadow queries was actually returning for purely index queries a lot actually when not bootstrapped, which is definitely incorrect since we don't have any of the data and only in memory (incomplete data).

Reply:
huh interesting...I guess is there a reason we can't use the bootstrap state on the namespace?

Reply:
When shards are added/removed and namespace needs another bootstrap, the bootstrap get's reset to "Bootstrapping" and we still need to return data at that point:
```go
n.bootstrapState = Bootstrapping
```